packages/blueprints/src/base/showstyle/executeActions/steppedGraphicExample.ts

```diff
@@ -0,0 +1,205 @@
+ *
+ * Provides example for defining the actionOptions schema and triggerModes
+ *
+ * @param ingestRundown - {ExtendedIngestRundown}
+ * @returns IBlueprintActionManifest
+ */
+export const exampleGraphicNextStepAdlibAction = (ingestRundown: ExtendedIngestRundown): IBlueprintActionManifest =>
+	literal<IBlueprintActionManifest>({
+		actionId: ActionId.GFXStep,
+		userData: {},
+		/** Describes the expected format of `userData` and `actionOptions` passed to the action when executed.
+		 * userData is currently never passed to the action, reserved for future functionality to allow users to change AdLib settings in the Sofie UI
+		 */
+		userDataManifest: {
+			optionsSchema: JSONBlobStringify({
+				$schema: 'http://json-schema.org/draft-04/schema#',
+				type: 'object',
+				properties: {
+					jumpTo: {
+						type: 'number',
+					},
+					increment: {
+						type: 'number',
+					},
+				},
+			}),
+		},
+		display: {
+			label: t('Control Graphic Step'),
+			sourceLayerId: SourceLayer.GFX,
+			outputLayerId: getOutputLayerForSourceLayer(SourceLayer.GFX),
+		},
+		triggerModes: [
+			{
+				data: ExampleGFXStepActionTriggerModes.NEXT,
+				display: {
+					_rank: 0,
+					label: t('Next Slide'),
+					description: t('this is a description'),
+				},
+			},
+			{
+				data: ExampleGFXStepActionTriggerModes.PREV,
+				display: {
+					_rank: 0,
+					label: t('Previous Slide'),
+					description: t('this is a description'),
+				},
+			},
+		],
+		externalId: ingestRundown.externalId,
+	})
+
+export enum ExampleGFXStepActionTriggerModes {
+	NEXT = 'next',
+	PREV = 'prev',
+}
+
+export interface ExampleGFXStepActionOptions {
+	increment: number
+	jumpTo: number | undefined
+}
+
+/**
+ * A global action that can be executed any-time in the show.
+ * We check if our current part has any stepped graphic pieces and step through them until we reach the last step.
+ *
+ * Also supports trigger modes to step backwards and actionOptions to jump to a specific step or change the increment.
+ *
+ * Be sure to add it to `executeAction` in 'packages/blueprints/src/base/showstyle/executeActions/index.ts' if you are making your own.
+ *
+ * @param context - {IActionExecutionContext}
+ * @param triggerMode - {string} - A simple string value that describes how the AdLib shoul
+ * @param actionOptions - {ExampleActionOptions}
+ */
+export async function executeGraphicNextStep(
+	context: IActionExecutionContext,
+	triggerMode?: string,
+	actionOptions: ExampleGFXStepActionOptions = {
+		increment: 1,
+		jumpTo: undefined,
+	}
+): Promise<void> {
+	// we filter for any stepped graphic piece
+	const pieceInstances = await context.getPieceInstances('current')
+	const steppedPieceInstances = pieceInstances.filter(
+		(piece) => (piece.piece.content as unknown as WithTimeline<NoraContent>).step
+	)
+
+	// we execute the action for each of them
+	for (const piece of steppedPieceInstances) {
+		const content = piece.piece.content as unknown as WithTimeline<NoraContent>
```

Comment:
You probably want to filter down to only pieces which are currently playing, in case there is a piece in the current part which has been stopped by adlibbing another piece.

We use something like:
```
    const now = context.getCurrentTime()
	const currentPieceInstances = (await context.getResolvedPieceInstances('current'))
		// Ensure the order is unchanged:
		.sort((a, b) => {
			if (a._id > b._id) return 1
			if (a._id < b._id) return -1
			return 0
		})
		.filter((pi) => {
			// Check that this piece is active
			if (pi.resolvedStart > now) return false
			if (pi.resolvedDuration) {
				const resolvedEnd = pi.resolvedStart + pi.resolvedDuration
				if (resolvedEnd < now) return false
			}
			return true
		})
```